Kernel/init.cpp

```diff
@@ -192,6 +192,38 @@ extern "C" {
 multiboot_info_t* multiboot_info_ptr;
 }
 
+// todo remove
+struct global_struct
+{
+    global_struct() : test_int(100) { kprintf("Globalglobalglobalglobalglobal\n"); }
+    global_struct(int val) : test_int(val) {}
+
+    ~global_struct() {}
+    int test_int;
+};
+
+// todo remove
+global_struct g_test_global;
+global_struct g_other_global(10);
+
+typedef void (*ctor_func_t)(); 
+
+// Defined in the linker script
+extern ctor_func_t start_ctors;
+extern ctor_func_t end_ctors;
+
+// Define some Itanium C++ ABI methods to stop the linker from complaining
+// If we actually call these somthing has gone horribly wrong
+void* __dso_handle __attribute__((visibility ("hidden")));
+
+extern "C" int __cxa_atexit ( void (*f)(void *), void *p, void *d )
+{
+    (void)f;
+    (void)p;
+    (void)d;
+    return 0;
```

Comment:
Let's put an `ASSERT_NOT_REACHED();` in this function.

Reply:
done, also removed parameter names